src/extensions/wwpdb/struct-conn/index.ts

```diff
@@ -0,0 +1,295 @@
+        result[conn.id] = conn;
+    }
+    return result;
+}
+
+/** Return MolScript expression for atoms or residues involved in a struct_conn */
+function structConnExpression(conn: StructConnRecord, by: 'atoms' | 'residues') {
+    const { core, struct } = MolScriptBuilder;
+    const partnerExpressions = [];
+    for (const partner of [conn.partner1, conn.partner2]) {
+        const propTests: Parameters<typeof struct.generator.atomGroups>[0] = {
+            'chain-test': core.rel.eq([struct.atomProperty.macromolecular.label_asym_id(), partner.asymId]),
+            'group-by': struct.atomProperty.core.operatorName(),
+        };
+        if (partner.seqId !== undefined) {
+            propTests['residue-test'] = core.rel.eq([struct.atomProperty.macromolecular.label_seq_id(), partner.seqId]);
+        } else if (partner.authSeqId !== undefined) { // for the case of water and carbohydrates (see 5elb, covale3 vs covale5)
+            propTests['residue-test'] = core.logic.and([
+                core.rel.eq([struct.atomProperty.macromolecular.auth_seq_id(), partner.authSeqId]),
+                core.rel.eq([struct.atomProperty.macromolecular.pdbx_PDB_ins_code(), partner.insCode]),
+            ]);
+        }
+        if (by === 'residues' && partner.altId !== '') {
+            propTests['atom-test'] = core.rel.eq([struct.atomProperty.macromolecular.label_alt_id(), partner.altId]);
+        }
+        if (by === 'atoms') {
+            propTests['atom-test'] = core.logic.and([
+                core.rel.eq([struct.atomProperty.macromolecular.label_atom_id(), partner.atomId]),
+                core.rel.eq([struct.atomProperty.macromolecular.label_alt_id(), partner.altId]),
+            ]);
+        }
+        partnerExpressions.push(struct.filter.first([struct.generator.atomGroups(propTests)]));
+    }
+    return struct.combinator.merge(partnerExpressions.map(e => struct.modifier.union([e])));
+}
+
+/** Create visuals for residues and atoms involved in a struct_conn and zoom on them */
+async function addStructConnInspection(plugin: PluginContext, structNode: StructNode, conn: StructConnRecord) {
+    const expressionByResidues = structConnExpression(conn, 'residues');
+    const expressionByAtoms = structConnExpression(conn, 'atoms');
+
+    const update = plugin.build();
+    update.to(structNode).apply(
+        StructureComponent,
+        { label: `${conn.id} (residues)`, type: { name: 'expression', params: expressionByResidues } },
+        { tags: [TAGS.RESIDUE_SEL] }
+    ).apply(
+        StructureRepresentation3D,
+        RESIDUES_VISUAL_PARAMS,
+        { tags: [TAGS.RESIDUE_REPR] }
+    );
+
+    const atomsVisRef = update.to(structNode).apply(
+        StructureComponent,
+        { label: `${conn.id} (atoms)`, type: { name: 'expression', params: expressionByAtoms } },
+        { tags: [TAGS.ATOM_SEL] }
+    ).apply(
+        StructureRepresentation3D,
+        ATOMS_VISUAL_PARAMS,
+        { tags: [TAGS.ATOM_REPR] }
+    ).ref;
```

Comment:
You can use `.selector` and then you dont have to do the `.select(atomsVisRef)[0]` below.

Reply:
Cool, thanks.